Add 1 on 1 video calling/client.js

```diff
@@ -0,0 +1,159 @@
+import { CallClient, CallAgent, Renderer, LocalVideoStream} from "@azure/communication-calling";
+import { AzureCommunicationTokenCredential } from '@azure/communication-common';
+
+let call;
+let callAgent;
+const calleeInput = document.getElementById("callee-id-input");
+const callButton = document.getElementById("call-button");
+const hangUpButton = document.getElementById("hang-up-button");
+const stopVideoButton = document.getElementById("stopVideo");
+const startVideoButton = document.getElementById("startVideo");
+
+let placeCallOptions;
+let deviceManager;
+let localVideoStream;
+let remoteVideoStream;
+let rendererLocal;
+let rendererRemote;
+
+function handleVideoStreams(v) {
+  remoteVideoStream = v;
+  remoteVideoView();
+  remoteVideoStream.on('availabilityChanged', async () => {
+    if (remoteVideoStream.isAvailable) {
+        remoteVideoView();
+    } else {
+        rendererRemote.dispose();
+    }
+  });
+  if (remoteVideoStream.isAvailable) {
+    remoteVideoView();
+  }
+}
+
+function subscribeToRemoteParticipant(p) {
+  p.videoStreams.forEach(v => {
+    handleVideoStreams(v);
+  });
+  p.on('videoStreamsUpdated', e => {
+    e.added.forEach(v => {
+      handleVideoStreams(v);
+    })
+  });
+}
+
+
+async function init() {
+  const callClient = new CallClient();
+  const tokenCredential = new AzureCommunicationTokenCredential("<USER ACCESS TOKEN>");
+  callAgent = await callClient.createCallAgent(tokenCredential, { displayName: 'optional ACS user name' });
+      
+  deviceManager = await callClient.getDeviceManager();
+  callButton.disabled = false;
+
+  callAgent.on('incomingCall', async e => {
+    const videoDevices = await deviceManager.getCameras();
+    const videoDeviceInfo = videoDevices[0];
+    localVideoStream = new LocalVideoStream(videoDeviceInfo);
+    localVideoView();
+
+    stopVideoButton.disabled = false;
+    callButton.disabled = true;
+    hangUpButton.disabled = false;
+
+    const addedCall = await e.incomingCall.accept({videoOptions: {localVideoStreams:[localVideoStream]}});
+    call = addedCall;
+
+    addedCall.remoteParticipants.forEach( p => {
+      subscribeToRemoteParticipant(p);
+    })
+
+    addedCall.on('remoteParticipantsUpdated', e => {
+      e.added.forEach( p => {
+        subscribeToRemoteParticipant(p);
+      })
+    });        
+  });
+
+  callAgent.on('callsUpdated', e => {
+    e.removed.forEach(removedCall => {
+      // dispose of video renderers
+      rendererLocal.dispose();
+      rendererRemote.dispose();
+      // toggle button states
+      hangUpButton.disabled = true;
+      callButton.disabled = false;
+      stopVideoButton.disabled = true;
+    })
+  })
+}
+init();
```

Comment:
do you want to await this also, is it ok to have this outer scope and call init from the root scope?

Reply:
Couldn't await init() since it's not inside any async function.